node-upstream-tests/run_tests.js

```diff
@@ -0,0 +1,387 @@
+'use strict'
+
+/* eslint-disable no-console */
+
+const promisify = require('util').promisify
+const axios = require('axios')
+const childProcess = require('child_process')
+const fs = require('fs')
+const glob = promisify(require('glob'))
+const path = require('path')
+
+const NODE_BIN = process.env['NODE_BIN'] || '/usr/bin/node'
+const NODE_REPO_PATH = process.env['NODE_PROJECT']
+if (NODE_REPO_PATH === undefined) {
+  throw new Error('The env variable NODE_PROJECT is not set. This is required to locate the root of the nodejs repo')
+}
+
+const NEEDS_TO_SPAWN_AGENT = false
+
+const MODULES = [
+  'dns',
+  'fs',
+  'tcp',
+  'http',
+  'http2',
+  'net'
+]
+
+// Taken from nodeJS repo.
+// These are not run by default on their CI because they require special setups
+const IGNORED_SUITES = [
+  'addons',
+  'benchmark',
+  'doctool',
+  'embedding',
+  'internet',
+  'js-native-api',
+  'node-api',
+  'pummel',
+  'tick-processor',
+  'v8-updates'
+]
+
+const TRACING_HEADERS = [
+  'x-datadog-trace-id',
+  'x-datadog-parent-id',
+  'x-datadog-sampled'
+]
+
+const UNEXPECTED_FAILURES = [
+  'test/parallel/test-dns-lookup-promises.js',
+  'test/parallel/test-dns-lookup.js',
+  'test/parallel/test-dns-lookupService.js',
+  'test/async-hooks/test-http-agent-handle-reuse-parallel.js',
+  'test/async-hooks/test-http-agent-handle-reuse-serial.js',
+  'test/parallel/test-http-client-check-http-token.js',
+  'test/parallel/test-http-invalid-urls.js',
+  'test/parallel/test-http-max-headers-count.js',
+  'test/parallel/test-http-parser-lazy-loaded.js',
+  'test/sequential/test-http2-timeout-large-write-file.js',
+  'test/parallel/test-net-connect-call-socket-connect.js',
+  'test/parallel/test-http2-padding-aligned.js',
+  'test/parallel/test-http-same-map.js',
+  'test/parallel/test-http-deprecated-urls.js',
+  'test/parallel/test-fs-access.js',
+  'test/parallel/test-fs-chmod.js',
+  'test/parallel/test-fs-chown-type-check.js',
+  'test/parallel/test-fs-close-errors.js',
+  'test/parallel/test-fs-copyfile.js',
+  'test/parallel/test-fs-error-messages.js',
+  'test/parallel/test-fs-fchmod.js',
+  'test/parallel/test-fs-fchown.js',
+  'test/parallel/test-fs-fsync.js',
+  'test/parallel/test-fs-lchmod.js',
+  'test/parallel/test-fs-lchown.js',
+  'test/parallel/test-fs-make-callback.js',
+  'test/parallel/test-fs-makeStatsCallback.js',
+  'test/parallel/test-fs-open.js',
+  'test/parallel/test-fs-opendir.js',
+  'test/parallel/test-fs-read.js',
+  'test/parallel/test-fs-realpath-native.js',
+  'test/parallel/test-fs-realpath.js',
+  'test/parallel/test-fs-stat.js',
+  'test/parallel/test-fs-truncate.js'
+]
+
+// These tests trigger a stackoverflow in the test agent because
```

Comment:
Is that unique to the test agent or would the real agent also crash under these conditions?